packages/core/src/tools/edit.ts

```diff
@@ -72,12 +73,7 @@ export class EditTool
   implements ModifiableTool<EditToolParams>
 {
   static readonly Name = 'replace';
-  private readonly rootDirectory: string;
```

Comment:
Instead of re-calling `config.getTargetDir()` it might be valuable to keep this and pass it through everywhere. Same for other tools. I'd have GC run through them on your behalf 😄